src/playercards/AllCardsBag.ttslua

```diff
@@ -131,70 +254,179 @@ function buildSupplementalIndexes()
-            cycleName = string.sub(cycleName, 11)
+      local checkedTraits = {}
+      if cardMetadata.traits then
+        for traitString in string.gmatch(cardMetadata.traits, "%w+.") do
+          local trait = string.lower(string.gsub(traitString, "[.]", ""))
+          checkedTraits[trait] = true
+          if traitIndex[trait] == nil then
+            traitIndex[trait] = { }
           end
-          if cycleName == "the night of the zealot" then
-            cycleName = "core"
+          table.insert(traitIndex[trait], cardMetadata.id)
+        end
+      end
+      if bonded then
+        for _, bondData in ipairs(bonded) do
+          local bondCardMetadata = cardIdIndex[bondData.id].metadata
+          for traitString in string.gmatch(bondCardMetadata.traits, "%w+.") do
+            local trait = string.lower(string.gsub(traitString, "[.]", ""))
+            if not checkedTraits[trait] then
+              if traitIndex[trait] == nil then
+                traitIndex[trait] = { }
+              end
+              table.insert(traitIndex[trait], cardMetadata.id)
+            end
           end
-          if cycleIndex[cycleName] == nil then
-            cycleIndex[cycleName] = { }
+        end
+      end
+
+      if cardMetadata.willpowerIcons then
+        if not iconIndex.willpower[cardMetadata.willpowerIcons] then
+          iconIndex.willpower[cardMetadata.willpowerIcons] = { }
+        end
+        table.insert(iconIndex.willpower[cardMetadata.willpowerIcons], cardMetadata.id)
+      else
+        table.insert(iconIndex.willpower[0], cardMetadata.id)
+      end
+      if cardMetadata.intellectIcons then
+        if not iconIndex.intellect[cardMetadata.intellectIcons] then
+          iconIndex.intellect[cardMetadata.intellectIcons] = { }
+        end
+        table.insert(iconIndex.intellect[cardMetadata.intellectIcons], cardMetadata.id)
+      else
+        table.insert(iconIndex.intellect[0], cardMetadata.id)
+      end
+      if cardMetadata.agilityIcons then
+        if not iconIndex.agility[cardMetadata.agilityIcons] then
+          iconIndex.agility[cardMetadata.agilityIcons] = { }
+        end
+        table.insert(iconIndex.agility[cardMetadata.agilityIcons], cardMetadata.id)
+      else
+        table.insert(iconIndex.agility[0], cardMetadata.id)
+      end
+      if cardMetadata.combatIcons then
+        if not iconIndex.combat[cardMetadata.combatIcons] then
+          iconIndex.combat[cardMetadata.combatIcons] = { }
+        end
+        table.insert(iconIndex.combat[cardMetadata.combatIcons], cardMetadata.id)
+      else
+        table.insert(iconIndex.combat[0], cardMetadata.id)
+      end
+      if cardMetadata.wildIcons then
+        if not iconIndex.wild[cardMetadata.wildIcons] then
+          iconIndex.wild[cardMetadata.wildIcons] = { }
+        end
+        table.insert(iconIndex.wild[cardMetadata.wildIcons], cardMetadata.id)
+      else
+        table.insert(iconIndex.wild[0], cardMetadata.id)
+      end
```

Comment:
This could be simplified too

Reply:
How would you suggest?  Unlike classes, all the icons are separate metadata entries, so I can't think of any way around manually checking for each one.